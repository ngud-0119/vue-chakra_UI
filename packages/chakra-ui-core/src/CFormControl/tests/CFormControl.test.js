import { CFormControl, CInput, CFormLabel } from '../..'
import { render } from '@/tests/test-utils'

const renderComponent = (props) => {
  const inlineAttrs = (props && props.inlineAttrs) || ''
  const base = {
    components: { CFormControl, CFormLabel, CInput },
    template: `
    <CFormControl ${inlineAttrs}>
      <CFormLabel for="fname">First name</CFormLabel>
      <CInput id="fname" placeholder="First name" />
    </CFormControl>`,
    ...props
  }
  return render(base)
}

it('should render correctly', () => {
  const { asFragment } = renderComponent()

  expect(asFragment()).toMatchSnapshot()
})
