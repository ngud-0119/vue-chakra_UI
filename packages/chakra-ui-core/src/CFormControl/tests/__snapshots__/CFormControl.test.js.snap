// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<DocumentFragment>
  .emotion-1 {
  font-size: var(--chakra-fontSizes-md);
  padding-right: 12px;
  padding-bottom: 4px;
  opacity: 1;
  font-weight: var(--chakra-fontWeights-medium);
  font-family: var(--chakra-fonts-body);
  text-align: left;
  vertical-align: middle;
  display: inline-block;
}

.emotion-2 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  -webkit-transition: all 0.2s;
  transition: all 0.2s;
  outline: 2px solid transparent;
  outline-offset: 2px;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  font-size: var(--chakra-fontSizes-md);
  -webkit-padding-start: var(--chakra-space-4);
  padding-inline-start: var(--chakra-space-4);
  -webkit-padding-end: var(--chakra-space-4);
  padding-inline-end: var(--chakra-space-4);
  height: var(--chakra-sizes-10);
  line-height: 2.5rem;
  border-radius: var(--chakra-radii-md);
  border-width: 1px;
  border-color: inherit;
  background: var(--chakra-colors-white);
  font-family: var(--chakra-fonts-body);
}

.emotion-2[aria-readonly=true],
.emotion-2[readonly],
.emotion-2[data-readonly] {
  background: var(--chakra-colors-transparent);
  box-shadow: none!important;
  -webkit-user-select: all;
  -moz-user-select: all;
  -ms-user-select: all;
  user-select: all;
}

.emotion-2:hover,
.emotion-2[data-hover] {
  border-color: var(--chakra-colors-gray-300);
}

.emotion-2[disabled],
.emotion-2[aria-disabled=true],
.emotion-2[data-disabled] {
  opacity: 0.4;
  cursor: not-allowed;
}

.emotion-2:focus,
.emotion-2[data-focus] {
  z-index: 1;
  border-color: #7db1ff;
  box-shadow: 0 0 0 1px #7db1ff;
}

.emotion-2[aria-invalid=true],
.emotion-2[data-invalid] {
  border-color: #e66673;
  box-shadow: 0 0 0 1px #e66673;
}

<div
    class="emotion-0"
    data-chakra-component="CFormControl"
    role="group"
  >
    <label
      class="emotion-1"
      data-chakra-component="CFormLabel"
      for="fname"
    >
      First name
    </label>
     
    <input
      class="emotion-2"
      data-chakra-component="CInput"
      id="fname"
      placeholder="First name"
    />
  </div>
</DocumentFragment>
`;
